src/pages/activity.js

```diff
@@ -85,28 +84,26 @@ export default function Activity() {
   }, []);
 
   useEffect(() => {
-    let _transfers = daitransfers.concat(mainnetTransfers)
-      // Filter ongoing private events
-      .filter(t => {
-        // Check if transfer belongs to a private event
-        const privateEvent = privateEvents.find(e => e.id === t.token.event.id)
-        if (privateEvent === undefined) return true;
-
-        // If it does, check if it is an ongoing private event
-        let evEndDate = new Date(privateEvent.end_date.replace(/-/g, ' ')).getTime()
-        let now = new Date().getTime()
-
-        // Show any transfers from finished private events
-        return evEndDate <= now;
-      })
-      // Sort from newer to older
-      .sort((a, b) => {
-        return b.timestamp - a.timestamp
-      })
-      // Only show a few
-      .slice(0, transferLimit)
-    setTransfers(_transfers)
-  }, [daitransfers, mainnetTransfers, privateEvents])
+    const setNewTransfers = async () => {
+      let transfersEventIds = daitransfers.map(t => t.token.event.id).concat(mainnetTransfers.map(t => t.token.event.id))
+      transfersEventIds = transfersEventIds.filter(onlyUnique)
+      const {items: publicEvents} = await getPaginatedEvents({event_ids: transfersEventIds, privateEvents: false})
+      let _transfers = daitransfers.concat(mainnetTransfers)
+          // Filter out private events
+          .filter(t => {
+            const publicEvent = publicEvents.find(e => e.id === parseInt(t.token.event.id))
+            return publicEvent !== undefined
+          })
+          // Sort from newer to older
+          .sort((a, b) => {
+            return b.timestamp - a.timestamp
+          })
+          // Only show a few
+          .slice(0, transferLimit)
+      setTransfers(_transfers)
+    }
+    setNewTransfers().then()
+  }, [daitransfers, mainnetTransfers])
```

Comment:
Not sure about business logic but if `daitransfers` and `mainnetTransfers ` never change, for example if the `useEffects` where they are set fails it won't trigger this `useEffetc`. Plus if both retrieve information correctly this code will be run twice.